### DISCUSSION TITLE
Allow list of transformations in a camera calibration.

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/rosys#130] [Ideas / Feature Requests]
Currently the extrinsics of a camera callibration only allow a single transformation. That's fine for cameras placed at fix positions in the world. But for cameras mounted somewhere on a mobile robot we need the ability to have a local robot coordinate system and update the position in the world while driving. It could also get even more nested if the mobile robot has an n-DOF robot arm with a camera mounting.

As @NiklasNeugebauer pointed out, we may want to look into adapting https://github.com/dfki-ric/pytransform3d to do such transformations.

**Comment by rodja** [comment]
Maybe it would be good to merge https://github.com/zauberzeug/rosys/pull/98 before implementing this. Or we do this first...

**Comment by falkoschindler** [comment]
Concept:

- extend `Calibration` class with `extrinsics: list[Pose3d] = []`
- remove `rotation` from `Intrinsics`
- thoughts about persistence:
	- volatile poses vs. hardware-specific constants
	- how to load shared references from persistence? (see #106)

**Comment by falkoschindler** [comment]
This feature has been implemented in PR #153.
